flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingTracking.java

```diff
@@ -142,9 +142,11 @@ public double getMaxRestartTimeSecondsOrDefault(Configuration conf) {
                 }
             }
         }
+        long restartTimeFromConfig = conf.get(AutoScalerOptions.RESTART_TIME).toSeconds();
+        long maxRestartTimeFromConfig = conf.get(AutoScalerOptions.MAX_RESTART_TIME).toSeconds();
         return maxRestartTime == -1
                 ? restartTimeFromConfig
```

Comment:
I think we should always cap the RESTART_TIME by the configured MAX_RESTART_TIME (if configured).

Reply:
@gyfora I'd like to hear your thoughts on this

Reply:
Synced with Max offline - we decided to rename the option to `TRACKED_RESTART_TIME_LIMIT` to make the scope clear.

Reply:
sounds good!